frontend/asset_mapper.rst

```diff
@@ -0,0 +1,1030 @@
+it. So, it would look for ``/assets/duck.js``. That URL *would* be correct,
+except that the ``duck.js`` file is versioned. Fortunately, AssetMapper sees that
+import and adds a mapping from ``/assets/duck.js`` to the correct, versioned
+filename. The result: importing ``./duck.js`` just works!
+
+Preloading and Initializing "app.js"
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In addition to the importmap, the ``{{ importmap() }}`` Twig function also renders
+an `es module shim`_ and a few other things, like a set of "preloads":
+
+.. code-block:: html
+
+    <link rel="modulepreload" href="/assets/app-4e986c1a2318dd050b1d47db8d856278.js">
+    <link rel="modulepreload" href="/assets/duck-1b7a64b3b3d31219c262cf72521a5267.js">
+
+In ``importmap.php``, each entry can have a ``preload`` option. If set to ``true``,
+a ``<link rel="modulepreload">`` tag is rendered for that entry as well as for
+any JavaScript files it imports (this happens for "relative" - ``./`` or ``../`` -
+imports only). This is a performance optimization and you can learn more about below
+in :ref:`Performance: Add Preloading <performance-preloading>`.
+
+.. _importmap-app-entry:
+
+The ``importmap()`` function also renders one more line:
+
+.. code-block:: html
+
+    <script type="module">import 'app';</script>
+
+So far, we've output an ``importmap`` and even hinted to the browser that it
+should preload some files. But we haven't actually told the browser to parse
+and execute any JavaScript. This line does that: it imports the ``app`` entry,
+which causes the code in ``assets/app.js`` to be executed.
+
+Importing Specific Files From a 3rd Party Package
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Sometimes you'll need to import a specific file from a package. For example,
+suppose you're integrating `highlight.js`_ and want to import just the core
+and a specific language:
+
+.. code-block:: javascript
+
+    import hljs from 'highlight.js/lib/core';
+    import javascript from 'highlight.js/lib/languages/javascript';
+
+    hljs.registerLanguage('javascript', javascript);
+    hljs.highlightAll();
+
+In this case, adding the ``highlight.js`` package to your ``importmap.php`` file
+won't work: whatever your importing - e.g. ``highlight.js/lib/core`` - needs to
+*exactly* match an entry in the ``importmap.php`` file.
+
+Instead, use ``importmap:require`` and pass it the exact paths you need. This
+also shows how you can require multiple packages at once:
+
+.. code-block:: terminal
+
+    $ php bin/console importmap:require highlight.js/lib/core highlight.js/lib/languages/javascript
+
+Handling 3rd-Party CSS
+----------------------
+
+With the ``importmap:require`` command, we can quickly use any JavaScript
+package. But what about CSS? For example, the ``bootstrap`` package also contains
+a CSS file.
+
+Including CSS is a bit more manual, but still easy enough. To find the CSS,
+we recommend using `jsdelivr.com`_:
```

Comment:
```suggestion
Including CSS is a bit more manual, but still easy enough. You need to find
a CDN that provides the required CSS files, for instance `jsdelivr.com`_:
```

Reply:
Is it because we want to avoid using `we`? I think the original sentence sounds better.

Reply:
Hmm. If you want to download the asset, you don't need a CDN: you could find the package anywhere - like on npm - you can browser the files - https://www.npmjs.com/package/bootstrap?activeTab=code and then download the one your want.

jsdelivr just seems to have the "best experience" for finding the files. WDYT?